### DISCUSSION TITLE
Clear all bans and ignores when nicotine starts

### DISCUSSION THREAD

**Discussion by ghost** [nicotine-plus/nicotine-plus#2257] [Q&A] [answered]
I've just seen that the Leech Detector that I have setup to ban leechers wasn't sending the message that I had configured because that was sent at the end of the upload and I was canceling the upload before banning, so I wasn't telling people why they were being banned.

I want to clear all bans and ignores when nicotine starts. How can I do that?

**Comment by mathiascode** [comment, marked as answer]
Run the following code in a plugin when it loads (e.g. in the `loaded_notification` method):

```
for list_name in ("banlist", "ignorelist", "ipblocklist", "ipignorelist"):
    self.config.sections["server"][list_name].clear()
```
